### PR TITLE
feat: make sym_n search for h_err and h_sp, or introduce new goals if they are not present

### PR DESCRIPTION
**Pull request by alexkeizer**
### Description:

Expands upon #69 to include context search for `h_err` and `h_sp`.
These assumptions are not strictly required, since we don't reflect information from them, 
so if they are not present we  introduce a new goal for them (in contrast to the other hypotheses, for which we simply throw an error).

### Testing:

`make all` succeeds, including conformance testing

### License:

By submitting this pull request, I confirm that my contribution is
made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by alexkeizer** [comment]
This PR depends on #69

**Comment by alexkeizer** [comment]
I've pulled in main after the dependecy got merged, so this is now ready for review: @bollu @shigoel

File: Tactics/SymContext.lean [hunk 1]
```diff
@@ -160,7 +217,7 @@ def fromLocalContext (state? : Option Name) : MetaM SymContext := do
   -- Find `h_run` and infer `runSteps` from it
   let sf ← mkFreshExprMVar none
   let runSteps ← mkFreshExprMVar (Expr.const ``Nat [])
-  let h_run_type ← mkEq sf (←mkAppM ``run #[runSteps, stateExpr])
+  let h_run_type ← mkEq sf (←mkAppM ``_root_.run #[runSteps, stateExpr])
```
Comment (shigoel) [inline Tactics/SymContext.lean:220]: This is the part where having namespaced `run` would've helped, I see.

**Review by shigoel — APPROVED** [review]
Nice! Thanks, @alexkeizer.

**Pull request merged**